site/en/docs/android/custom-tabs/guide-warmup-prefetch/index.md

```diff
@@ -0,0 +1,106 @@
+seoTitle: "Guide: warm-up and pre-fetching - using the Custom Tabs Service"
+date: 2023-04-21
+description: Learn how to make sure that your Custom Tabs start as quickly as possible.
+authors:
+  - sebastianbenz
+---
+
+The third part of this guide focuses on warming up the browser process via [warmup()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#warmup(long)) and prefetching web pages via [mayLaunchUrl](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsSession#mayLaunchUrl(android.net.Uri,android.os.Bundle,java.util.List%3Candroid.os.Bundle%3E))(). Warming up the browser process can save up to 700ms when opening a link. Pre-rendering content via mayLaunchUrl will make external content open instantly. Together both APIs can greatly improve the user experience of a Custom Tabs integration and are highly recommended.
+
+The basic steps are:
+
+1. Check via [CustomTabsClient](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient).[getPackageName()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#getPackageName(android.content.Context,java.util.List%3Cjava.lang.String%3E)) If the default browser supports Custom Tabs. If yes, bind to the CustomTabsService via [CustomTabsClient.bindCustomTabsService()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#bindCustomTabsService(android.content.Context,java.lang.String,androidx.browser.customtabs.CustomTabsServiceConnection)).
+2. Once connected to the CustomTabsService, in the [CustomTabsServiceConnection.onCustomTabsServiceConnected()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsServiceConnection#onCustomTabsServiceConnected(android.content.ComponentName,androidx.browser.customtabs.CustomTabsClient)) callback, do:
+    1. Warmup the browser process via [CustomTabsClient.warmup()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#warmup(long)).
+    2. Create a new [CustomTabsSession](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsSession) via [CustomTabsClient.newSession()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#newSession(androidx.browser.customtabs.CustomTabsCallback,int)).
+3. Optionally, prefetch web pages the user is likely to visit via [CustomTabsSession.mayLaunchUrl](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsSession#mayLaunchUrl(android.net.Uri,android.os.Bundle,java.util.List%3Candroid.os.Bundle%3E))().
+4. When launching a new Custom Tab, pass the [CustomTabsSession](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsSession) to the [CustomTabsIntent.Builder](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsIntent.Builder) via the constructor `new CustomTabsIntent.Builder(session)`.
```

Comment:
Personally, these steps as comments throughout the code rather than a separate section would be more helpful

Reply:
yeah, I added this because you can have links